aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt

```diff
@@ -21,17 +21,28 @@ import kotlinx.serialization.Serializable
 
 @Serializable
 sealed class AmplifyCredential {
+
+    interface UserPoolData {
```

Comment:
nit: should we name this as UserPoolTypeCredential and IdentityPoolTypeCredential?